src/FizzBuzz/FizzBuzz.php

```diff
@@ -0,0 +1,75 @@
+<?php
+
+namespace App\FizzBuzz;
+
+class FizzBuzz
+{
+    public function printFizzOnMultiplesOfThree(string $integers) : string
+    {
+        return implode(',', array_map(function ($int) {
+            if ($int % 3 === 0) {
+                return 'Fizz';
+            }
+
+            return $int;
+        }, explode(',', $integers)));
+    }
+
+    public function printBuzzOnMultiplesOfFive(string $integers) : string
+    {
+        return implode(',', array_map(function ($integer) {
+            if ($integer % 5 === 0) {
+                return 'Buzz';
+            }
+
+            return $integer;
+        }, explode(',', $integers)));
+    }
+
+    public function printFizzBuzzOnMultiplesOfThreeAndFive(string $integers) : string
+    {
+        return implode(',', array_map(function($integer) {
+            if ($integer % 15 === 0) {
+                return 'FizzBuzz';
+            }
+
+            return $integer;
+        }, explode(',', $integers)));
+    }
+
+    public function fizzBuzz(string $integers) : string
+    {
+        $this->stringContainsOnlyCommaSeparatedIntegers($integers);
+
+        return implode(',', array_map(function($integer) {
```

Comment:
Por conveniência você faz um `implode` e `explode` por vírgula aqui. O explode: faz um lookup por vírgula e aloca um array pra você iterar. O implode: itera um array e gera uma nova string por vírgula.

Isto é uma potencial micro otimização, mas acho importante criar esse senso de o que acontece por debaixo dos panos.

**Recomendo a leitura de duas funções:**
[`php_explode`](https://github.com/php/php-src/blob/master/ext/standard/string.c#L1054) a implementação interna do core.

[`php_implode`](https://github.com/php/php-src/blob/master/ext/standard/string.c#L1169) a implementação interna do core.

Dá uma olhada no que essas funções fazem por baixo dos panos. Foca no número de `loops` e de `malloc`. Via de regra `malloc` é caro pra fazer e quanto mais loops, maior o fator de tempo numa função.

**Implementações alternativas podem incluir:**
- `for loop` + `acumulador` + `regex`
- `regex` que extrai valores

Recomendo tentar cada uma delas e coletar timings :)